db/migrations/20230207080945_sub_categories.sql

```diff
@@ -0,0 +1,10 @@
+-- migrate:up
+CREATE TABLE product_sub_categories (
+    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    sub_category_name VARCHAR(20),
+    product_id INT,
+    CONSTRAINT product_sub_categories_product_id_fkey FOREIGN KEY (product_id) REFERENCES products (id)
+);
```

Comment:
- 제품을 포함하고 있는 최상단의 카테고리는 sub_categories라는 테이블명으로 수정하는 것이 좋을 것 같습니다.
- sub_category_name으로 컬럼을 짓게 되면 sub_categories.sub_category_name 과 같이 sub_category라는 단어가 중복되기 때문에 `name`으로 지어도 좋을 것 같습니다.
- 테이블 간 관계를 파악하면, sub_categories 테이블과 products 테이블은 1:N 관계입니다. 따라서 sub_categories 테이블이 products 테이블을 참조하는 것이 아니라,그 반대가 되어야 합니다.
- 추가로, main_category와 sub_category의 관계는 1:N이기 때문에 해당 테이블에서 main_categories 테이블을 참조해야합니다.
```suggestion
CREATE TABLE sub_categories (
    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
    name VARCHAR(20),
    main_category_id INT,
    CONSTRAINT sub_categories_main_category_id_fkey FOREIGN KEY (main_category_id) REFERENCES main_categories (id)
);
```